test/test_edit.py

```diff
@@ -0,0 +1,172 @@
+    external editor is simulated using `ModifyFileMocker`.
+    """
+    ALBUM_COUNT = 1
+    TRACK_COUNT = 10
+
+    def setUp(self):
+        self.setup_beets()
+        self.load_plugins('edit')
+        # make sure that we avoid invoking the editor except for making changes
+        self.config['edit']['diff_method'] = ''
+        # add an album
+        self.add_album_fixture(track_count=self.TRACK_COUNT)
+        # keep track of write()s
+        library.Item.write = Mock()
+
+    def tearDown(self):
+        self.teardown_beets()
+        self.unload_plugins()
+
+    def run_mocked_command(self, modify_file_args={}, stdin=[]):
+        """Run the edit command, with mocked stdin and yaml writing."""
+        m = ModifyFileMocker(**modify_file_args)
+        with patch.object(EditPlugin, 'get_editor', side_effect=m.action):
+            with control_stdin('\n'.join(stdin)):
+                self.run_command('edit')
+
+    def assertCounts(self, album_count=ALBUM_COUNT, track_count=TRACK_COUNT,
+                     write_call_count=TRACK_COUNT, title_starts_with=''):
+        """Several common assertions on Album, Track and call counts."""
+        self.assertEqual(len(self.lib.albums()), album_count)
+        self.assertEqual(len(self.lib.items()), track_count)
+        self.assertEqual(library.Item.write.call_count, write_call_count)
+        self.assertTrue(all(i.title.startswith(title_starts_with)
+                            for i in self.lib.items()))
+
+    def test_title_edit_discard(self):
+        """Edit title for all items in the library, then discard changes-"""
+        # edit titles
+        self.run_mocked_command({'replacements': {u't\u00eftle':
+                                                  u'modified t\u00eftle'}},
+                                # edit? y, done? y, modify? n
+                                ['y', 'y', 'n'])
+
+        self.assertCounts(write_call_count=0,
+                          title_starts_with=u't\u00eftle')
+
+    def test_title_edit_apply(self):
+        """Edit title for all items in the library, then apply changes."""
+        # edit titles
+        self.run_mocked_command({'replacements': {u't\u00eftle':
+                                                  u'modified t\u00eftle'}},
+                                # edit? y, done? y, modify? y
+                                ['y', 'y', 'y'])
+
+        self.assertCounts(write_call_count=self.TRACK_COUNT,
+                          title_starts_with=u'modified t\u00eftle')
+
+    def test_noedit(self):
+        """Do not edit anything."""
+        # do not edit anything
+        self.run_mocked_command({'contents': None},
+                                # edit? n -> "no changes found"
+                                ['n'])
+
+        self.assertCounts(write_call_count=0,
+                          title_starts_with=u't\u00eftle')
+
+    def test_malformed_yaml_discard(self):
```

Comment:
Yes, great point.

Ideally, I would event like something like a git commit: if you don't make any changes or just leave it malformed, the process immediately aborts. If you want to pick up where you left off, maybe we can provide a separate way to do that.